dali/kernels/transpose/transpose_gpu_impl_test.cu

```diff
@@ -190,6 +189,111 @@ TEST(TransposeTiled, BuildDescVectorized) {
   }
 }
 
+TEST(TransposeTiled, BuildDescAndForceMisalignment) {
+  TensorShape<> shape = { 57, 37, 53, 4 };  // a bunch of primes, just to make it harder
+  int size = volume(shape);
+  vector<uint8> in_cpu(size), out_cpu(size);
+  vector<uint8> ref(size);
+    
+  DeviceBuffer<uint8> in_gpu, out_gpu;
+  in_gpu.resize(size);
+  out_gpu.resize(size);
+    
+  for(uintptr_t offset = 0; offset < 4; offset++)
+  {
+    shape = { 57, 37, 53, 4 };
+    size = volume(shape);
+    std::iota(in_cpu.begin(), in_cpu.end(), 0);
+    CUDA_CALL(cudaMemset(out_gpu, 0xff, size*sizeof(int)));
+    shape = { 57, 37, 52, 4 };
```

Comment:
Do we need two shapes that differ by 1 in one of the extents of just using `shape = { 57, 37, 53, 4 }` and the allocation size `volume({57, 37, 53, 4 }) + 4` would be ok?

Reply:
changed